Tests/ApolloCodegenTests/CodeGeneration/Templates/TemplateString_DeprecationMessage_Tests.swift

```diff
@@ -0,0 +1,230 @@
+import XCTest
+import Nimble
+@testable import ApolloCodegenLib
+
+final class TemplateString_DeprecationMessage_Tests: XCTestCase {
+
+  let config = ApolloCodegen.ConfigurationContext(config: .mock(
+    options: .init(
+      warningsOnDeprecatedUsage: .include
+    )
+  ))
+
+  // MARK: - Swift @available Attribute Tests
+
+  func test__availableAttribute__givenSingleLineDeprecationMessageWithInnerDoubleQuotes_shouldEscapeDoubleQuotes() throws {
+    // given
+    let subject = TemplateString("""
+      \(deprecationReason: "not supported, use \"another thing\" instead.", config: config)
+      """)
+
+    let expected = #"""
+      @available(*, deprecated, message: "not supported, use \"another thing\" instead.")
+      """#
+
+    // then
+    let actual = subject.description
+
+    expect(actual).to(equalLineByLine(expected))
+  }
+
+  func test__availableAttribute__givenMultiLineDeprecationMessageWithInnerDoubleQuotes_shouldEscapeDoubleQuotes() throws {
+    // given
+    let subject = TemplateString("""
+      \(deprecationReason: "not supported\nuse \"another thing\" instead.", config: config)
+      """)
+
+    let expected = #"""
+      @available(*, deprecated, message: """
+        not supported
+        use \"another thing\" instead.
+        """)
+      """#
+
+    // then
+    let actual = subject.description
+
+    expect(actual).to(equalLineByLine(expected))
+  }
+
+  // MARK: Swift #warning Directive Tests
+
+  func test__warningDirective__givenSingleLineDeprecationMessageWithInnerDoubleQuotes_shouldEscapeDoubleQuotes() throws {
+    // given
+    let subject = TemplateString("""
+      \(field: "fieldOne", argument: "argOne", warningReason: "not supported, use \"another thing\" instead.")
+      """)
+
+    let expected = #"""
+      #warning("Argument 'argOne' of field 'fieldOne' is deprecated. Reason: 'not supported, use \"another thing\" instead.'")
+      """#
+
+    // then
+    let actual = subject.description
+
+    expect(actual).to(equalLineByLine(expected))
+  }
+
+  // MARK: SDL-to-Generation Test
+  //
+  // These tests ensure that when given double quotes in SDL the generation of Swift code works as
+  // expected from frontend parsing all the way to rendering of attributes, comments andwarnings.
+  // There is a test here for all the places that the GraphQL schema supports the @deprecated
+  // directive.
```

Comment:
These tests don't include the characters `\0` (null character), nor `\'` (single quote) because they're caught as syntax errors by graphql-js.
> failed: caught error: "Syntax Error: Invalid character escape sequence: "\0".

> failed: caught error: "Syntax Error: Invalid character escape sequence: "\'".

I think this is because graphql-js is validating strings against the [GraphQL spec definition of allowed escape characters](https://spec.graphql.org/draft/#sec-String-Value). Those two characters are not in that list.